src/hooks/privateChats.ts

```diff
@@ -44,6 +49,60 @@ export const usePrivateChatMessages = () => {
   useConnectPrivateChatMessages();
 
   const privateChatMessages = useSelector(privateChatMessagesSelector);
+  const { user } = useUser();
+  const userId = user?.uid;
+
+  const { chatSettings } = useChatSidebarControls();
+
+  const [play] = useSound(NEW_MESSAGE_SOUND, {
+    // `interrupt` ensures that if the sound starts again before it's
+    // ended, it will truncate it. Otherwise, the sound can overlap.
+    interrupt: true,
+  });
+
+  const [privateChatMessagesState, setPrivateChatMessagesState] = useState(
+    privateChatMessages
+  );
+
+  const chatWithUser =
+    chatSettings.openedChatType === ChatTypes.PRIVATE_CHAT
+      ? chatSettings.recipientId
+      : undefined;
+
+  useEffect(() => {
+    const newMessage = privateChatMessages?.filter(
+      ({ id: id1 }) =>
+        !privateChatMessagesState?.some(({ id: id2 }) => id2 === id1)
+    );
+
+    if (privateChatMessagesState === undefined) {
+      setPrivateChatMessagesState(privateChatMessages);
+    }
+
+    if (
+      privateChatMessages !== undefined &&
+      privateChatMessagesState !== undefined &&
+      newMessage !== undefined &&
+      newMessage.length !== 0 &&
+      privateChatMessages.length !== privateChatMessagesState.length &&
+      !newMessage[0].isRead &&
+      newMessage[0].from !== userId
+    ) {
+      setPrivateChatMessagesState(privateChatMessages);
+
+      if (chatWithUser === undefined) {
+        play();
+      } else if (chatWithUser !== newMessage[0].from) {
+        play();
+      }
+    }
+  }, [
+    privateChatMessages,
+    chatWithUser,
+    play,
+    privateChatMessagesState,
+    userId,
+  ]);
```

Comment:
The relevant code related to the notification sounds should be wrapped up in a custom hook (eg. `src/hooks/useNotificationSound.ts`), similar to how the previous PR was attempting to go (though properly using `useSound` and similar as per this PR):

- https://github.com/sparkletown/sparkle/pull/1180/files#diff-a22d769c39c4171c61bd19fa96454692ce900a294f3507e3c841a23086b17e47